### PR TITLE
Use Phoenix.Token for auth

### PR DESCRIPTION
**Pull request by mzgajner**

### CODE REVIEW

File: backend/lib/kjer_si/accounts.ex [hunk 1]
```diff
@@ -76,7 +76,7 @@ defmodule KjerSi.Accounts do
   """
   def get_user_with_preload(uid, preload) do
```
Comment (muki) [inline backend/lib/kjer_si/accounts.ex]: If it's id, let it be id. :)

```suggestion
  def get_user_with_preload(id, preload) do
```

File: backend/lib/kjer_si/accounts.ex [hunk 2]
```diff
@@ -76,7 +76,7 @@ defmodule KjerSi.Accounts do
...
     uid
```
Comment (muki) [inline backend/lib/kjer_si/accounts.ex]:
```suggestion
    id
```

File: backend/lib/kjer_si_web/controllers/helpers/accounts_helpers.ex [hunk 3]
```diff
@@ -3,41 +3,62 @@ defmodule KjerSi.AccountsHelpers do
...
+    if match do
+      {:ok, match["token"]}
+    else
+      {:error, :invalid_auth_header}
+    end
+  end
+
+  defp get_user_id(conn) do
+    with {:ok, token} <- get_token(conn) do
+      case Phoenix.Token.verify(KjerSiWeb.Endpoint, "user salt", token, max_age: 86400) do
```
Comment (muki) [inline backend/lib/kjer_si_web/controllers/helpers/accounts_helpers.ex]: We are not using per-user salts, so let's name it `universal salt` or even put it in an ENV.

File: backend/lib/kjer_si_web/controllers/helpers/accounts_helpers.ex [hunk 4]
```diff
@@ -3,41 +3,62 @@ defmodule KjerSi.AccountsHelpers do
...
+      invalid_user: "User not found",
+      unauthorized: "Not authorized to perform action",
+      not_found: "Resource not found",
+    }
 
-  def return_not_found(conn) do
     conn
       |> put_resp_content_type("application/json")
-      |> send_resp(:not_found, "{\"error\": \"not found\"}")
+      |> send_resp(400, "{\"error\": \"#{messages[type]}\"}")
```
Comment (muki) [inline backend/lib/kjer_si_web/controllers/helpers/accounts_helpers.ex]: Can we make not_found return 404?

File: backend/lib/kjer_si_web/controllers/helpers/accounts_helpers.ex (hunk 3 above)
Comment (mzgajner) [reply backend/lib/kjer_si_web/controllers/helpers/accounts_helpers.ex]: Yeah, good point, `"user salt"` is a tad misleading. I changed it to `"user auth"`, which is an example directly from [the docs](https://hexdocs.pm/phoenix/Phoenix.Token.html#module-example). Moving it to an env variable seems like overkill, it doesn't really increase entropy any more than just having a longer key in this case, so I'd prefer we only have one crypto env variable to worry about :)

File: backend/lib/kjer_si_web/controllers/helpers/accounts_helpers.ex (hunk 4 above)
Comment (mzgajner) [reply backend/lib/kjer_si_web/controllers/helpers/accounts_helpers.ex]: Added a few others that make sense.

**Pull request merged**